### DISCUSSION TITLE
Video Downloading

### DISCUSSION THREAD

**Discussion by abouddi** [AxisCommunications/discussions#67] [Axis Windows SDKs] [answered]
Several of our customers are facing a serious issue downloading videos from the camera using our software where we use the AxisMediaParser.

We are using the AxisMediaParser (axrtsp) to download the videos in “asf” format. Unfortunately, this is not working at some of our customers and we don't know the real cause of it.

We have tried to download the same videos using a normal http client and that is working. When we do this we can only download the videos in “mkv” format. Unfortunately, using this way we get “mkv” file that is missing the timeline of the video in which makes it impossible to step forward and backward in the video.

There are libraries to do conversion between different video formats and the output is satisfactory but unfortunately there is a licensing issues with that.

Is there any solution for this? Is there any way that we can download in “mkv” format and later run the “AxisMediaParser” on the locally stored video files (after being downloaded) to get the "asf" format?

**Comment by anders-axis** [comment, marked as answer]
> Is there any way that we can download in “mkv” format and later run the “AxisMediaParser” on the locally stored video files (after being downloaded) to get the "asf" format?
Yes, this should be possible, but you would need to deploy also the AXIS Matroska File Splitter (AxMatroskaSplitter.ax).

I would recommend creating a case with Axis help desk regarding the problems with downloads and provide details about under what circumstances this problem occur.

**Reply by abouddi** [reply]
I am not sure if the AxMatroskaSplitter.ax works for us as we want to do the conversion in our code and as I can see it is a plugin for windows media player. I will try with Axis help desk

**Reply by anders-axis** [reply]
If you install register AxMatroskaSplitter.ax, AMP will be able to load MKV-files in the same way as from a RTSP-stream using the MediaURL property.

**Reply by abouddi** [reply]
I will have to check this. Thank you
